lsp-mode.el

```diff
@@ -5167,6 +5164,17 @@ It will show up only if current point has signature help."
       (funcall action-handler action)
     (lsp--send-execute-command command arguments?)))
 
+(defun lsp-execute-command (_server command arguments)
```

Comment:
It seems to be that this does not improve backward compatibility

Reply:
It does. Many places call it as a function. I didn't actually find any other unfixed place that uses `cl-defmethod` on this across the entirety of github, only lots of elpa directories containing outdated ccls.elc? and lsp-java-*.elc? files.

Reply:
ok, this will work, but this is not the usecase covered by this method. It was an extension point, not an entry point supposed to be called.

Reply:
That is true, however people abused it like that anyway. Do you consider it acceptable to break all that code? (I actually like breaking changes, but we need to be careful for a project this large) There is no way I can go trough 800 projects to make this slightly more elegant.

Reply:
I think we should either clean it up or make sure it is fully backward compatible.

Reply:
Now I'm unsure: I can't seem to find anyone really using it: [github search](https://github.com/search?q=lsp-execute-command+-filename%3Accls.el+-filename%3Accls.elc+-filename%3Alsp-*&type=Code)

This kak-lsp stuff seems to be unrelated to lsp-mode, and the one use as a keybinding is wrong anyway, since that function wasn't ever interactive.

Reply:
@yyoncho by clean up, do you mean remove it? From my latest search, it seems that no one actually uses it.

Reply:
Can we do this in backward compatible way?

Maybe something like
``` elisp
(defvar lsp--from-internal-call nil)
(make-obsolete 'lsp-execute-command 'new-one "7.2")
(cl-defmethod lsp-execute-command (_server command arguments)
  "Dispatch COMMAND execution."
  (if lsp--from-internal-call
      :obsoleted
    (lsp--execute-command (lsp-make-command :command (symbol-name command)
                                            :arguments? arguments)))

(defun lsp-lens--create-interactive-command (command?)
  ;; ...
  (let ((lsp--from-internal-call t))
    (when (eq (lsp-execute-command server-id
                                     (intern (lsp:command-command command?))
                                     (lsp:command-arguments? command?))
                :obsoleted)
      (lsp--execute-command ...))))
```

Reply:
Either this or binding the full command in `lsp-lens--...` and checking if it is bound in lsp--execute-command. Not sure which one is less ugly.

Reply:
@yyoncho so you mean something like `fboundp` `lsp-execute-command` and leaving the latter undefined in `lsp-mode.el`? This would almost work, since `cl-defmethod` introduces the function if it doesn't yet exist. However, it would break code that calls `lsp-execute-command` as a function.

Reply:
No. The idea of @kiennq  is to make lsp-execute-command work as it is and still pass the title. The @kiennq's proposal can be achieved with some dynamic stope tricks. I am not 100% it worths the effort.

Reply:
I'll return it with an obsoletion warning, since I just had the following idea: we can leverage `cl-no-applicable-method`, limiting the amount of ugly hacks. It should still be removed at some point, perhaps in the next major release.